Tup#89

gdalyy opened the pull request:
Migrate old spaces data into blockchain

gdalyy committed:
Enabling manually ethereum for existing spaces

gdalyy committed:
fix add member call for already existing spaces

craxrev reviewed (approved):
looks good :+1:

bumi commented on calls/Space.php:
```diff
@@ -161,4 +169,59 @@ public static function leaveSpace($event)
+            $mintEvent = new Event(['sender' => $transaction]);
+
+            Coin::mintCoin($mintEvent);
+        }
```
I am wondering if this reflects the current coin status or only the issuing? 

does the current coin holder hold the coins on the smart contract in the end?

gdalyy replied:
@bumi sorry i didn't understand your questions , through this code block , i'm doing just `mint()` for already issued coin but which are not present on the smart contracts. 
Should i take into consideration , transfer transactions too ? which means that after the `mint()` action , i should loop on transfer transactions and call POST /transfer ?

bumi replied:
we have to make sure that the current balances of all users are also correct. 
if we only mint the coins then the balances don't match, do they? 

so we should mint coins for each current holder of those coins. 

//cc @haythem96

haythemsellami replied:
Ahh yeah, we need to make sure that the balances of the coin holders in the smart contract is the same as in humhub DB, so Instead of minting coins for the issue transaction and then also for the transfer... why not just fetching the coin holders balance and mint for each one of them ? ... what do u think @gdalyy

gdalyy replied:
You're right , i'll do the transfer so we make sure that balances on humhub are the same on the blockchain.

gdalyy replied:
One thing to mention : i'll be forcing wallet creation for users that do not belong to the space in question and which are holders of the space coin.

gdalyy replied:
Done

gdalyy committed:
transfer transactions

bumi commented on calls/Space.php:
```diff
@@ -161,4 +169,62 @@ public static function leaveSpace($event)
+            } else {
+                //transfer coins for each coin holder
+                Coin::transferCoin($transactionEvent);
+            }
```
I would not do all transfers but only call a `mintCoin` for each owner of an asset with the user's current balance? 
this is probably more correct but could lead to many ethereum transactions.
but as probably do not have that many transfers it might be fine?

gdalyy replied:
I think even with transfer calls this will be fine , because as you said currently we don't have many transfers to do.

gdalyy committed:
adding Authentication header for every ethereum http call

gdalyy committed:
- removing details call (GET /space)
- removing issue coin (POST /coin/issue) -> included in dao creation (POST /dao)
- remove wallet creation for accounts TYPE_ISSUE

gdalyy committed:
quick fix for coin symbol
